packages/core/src/Services/socket-general.js

```diff
@@ -213,17 +213,15 @@ const BinarySocketGeneral = (() => {
                 // DBot handles this internally. Special case: 'client.invalid_token'
                 if (active_platform === 'DBot') return;
 
-                client_store.logout().then(() => {
-                    let redirect_to = routes.trade;
-                    const action = getActionFromUrl();
-                    if (action === 'system_email_change') {
-                        return;
-                    }
-                    if (active_platform === 'Deriv MT5') {
-                        redirect_to = routes.mt5;
-                    }
-                    common_store.routeTo(redirect_to);
-                });
+                if (getActionFromUrl() !== 'system_email_change') {
+                    client_store.logout().then(() => {
+                        let redirect_to = routes.trade;
+                        if (active_platform === 'Deriv MT5') {
+                            redirect_to = routes.mt5;
+                        }
+                        common_store.routeTo(redirect_to);
+                    });
+                }
```

Comment:
Does it throw an `InvalidToken` error when we visit the confirmation link?

Reply:
yes it does, after clicking the email verification link

Reply:
but why?